core/trino-spi/src/main/java/io/trino/spi/connector/UnsupportedViewDialectException.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import io.trino.spi.TrinoException;
+
+import static io.trino.spi.StandardErrorCode.UNSUPPORTED_VIEW_DIALECT;
+import static java.util.Objects.requireNonNull;
+
+public class UnsupportedViewDialectException
```

Comment:
This exception was introduced to be able to surface meaningful failures for the following case.
a user creates a view `spark_view` via Spark.
The user goes to Trino and then tries to do "show create spark_view" or "describe" or "drop".

Before this change, we would fail with a "No such view" exception since we strict resolve a Trino dialect. That's not entirely correct since the view does exist but with a different dialect and Trino should ideally surface those details (i.e which dialect) in the failure. @findepi